shell/platform/darwin/ios/framework/Source/FlutterPlatformPlugin.mm

```diff
@@ -0,0 +1,174 @@
+// Copyright 2016 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/shell/platform/darwin/ios/framework/Source/FlutterPlatformPlugin.h"
+
+#include <AudioToolbox/AudioToolbox.h>
+#include <Foundation/Foundation.h>
+#include <UIKit/UIApplication.h>
+#include <UIKit/UIKit.h>
+
+static NSDictionary* GetDirectoryOfType(NSSearchPathDirectory dir) {
+  NSArray* paths =
+      NSSearchPathForDirectoriesInDomains(dir, NSUserDomainMask, YES);
+  if (paths.count == 0)
+    return nil;
+  return @{ @"path": paths.firstObject };
+}
+
+namespace flutter {
+namespace platform {
+
+// TODO(abarth): Move these definitions from system_chrome_impl.cc to here.
+extern const char* const kOrientationUpdateNotificationName;
+extern const char* const kOrientationUpdateNotificationKey;
+extern const char* const kOverlayStyleUpdateNotificationName;
+extern const char* const kOverlayStyleUpdateNotificationKey;
+
+}  // namespace platform
+}  // namespace flutter
+
+using namespace flutter::platform;
+
+@implementation FlutterPlatformPlugin
+
+@synthesize messageName = _messageName;
+
+- (instancetype) init {
+  self = [super init];
+  if (self)
+    self->_messageName = @"flutter/platform";
+  return self;
+}
+
+- (NSDictionary*)didReceiveJSON:(NSDictionary*)message {
+  NSString* method = [message objectForKey:@"method"];
+  NSArray* args = [message objectForKey:@"args"];
+  if ([method isEqualToString:@"SystemSound.play"]) {
+    [self playSystemSound:[args objectAtIndex:0]];
```

Comment:
`[args objectAtIndex:0]` will throw an exception if there are no items in the array. `firstObject` returns nil instead in such cases if that if more suitable for this use case. In any case, we should probably not throw ObjC exceptions.